HCPCS/manual work/hcpcs_refresh.sql

```diff
@@ -0,0 +1,126 @@
+        update text;
+    BEGIN
+        SELECT TO_CHAR(CURRENT_DATE, 'YYYY_MM_DD')
+        INTO update;
+        EXECUTE FORMAT('create table %I as select * from concept_manual',
+                       'concept_manual_backup_' || update);
+
+    END
+$body$;
+
+--restore concept_manual table (run it only if something went wrong)
+/*TRUNCATE TABLE dev_hcpcs.concept_manual;
+INSERT INTO dev_hcpcs.concept_manual
+SELECT * FROM dev_hcpcs.concept_manual_backup_2022_07_26;*/
+
+--8.2.2. Create hcpcs_mapped table and pre-populate it with the resulting manual table of the previous HCPCS refresh.
+--DROP TABLE dev_hcpcs.hcpcs_mapped;
+/* CREATE TABLE dev_hcpcs.hcpcs_mapped
+(
+    id SERIAL PRIMARY KEY,
+    source_code_description varchar(255),
+    source_code varchar(50),
+    source_concept_class_id varchar(50),
+    source_invalid_reason varchar(20),
+    source_domain_id varchar(50),
+    to_value varchar(50),
+    source varchar(50),
+    target_concept_id int,
+    target_concept_code varchar(50),
+    target_concept_name varchar(255),
+    target_concept_class_id varchar(50),
+    target_standard_concept varchar(20),
+    target_invalid_reason varchar(20),
+    target_domain_id varchar(50),
+    target_vocabulary_id varchar(50)
+);*/
+
+--8.2.4. Truncate the hcpcs_mapped table. Save the spreadsheet as the hcpcs_mapped table and upload it into the working schema.
+/*TRUNCATE TABLE dev_hcpcs.hcpcs_mapped;*/
+
+--8.2.5. Deprecate all mappings that differ from the new version of resulting mapping file.
+UPDATE dev_hcpcs.concept_relationship_manual
+SET invalid_reason = 'D',
+    valid_end_date = current_date
+WHERE (concept_code_1, concept_code_2, relationship_id, vocabulary_id_2) IN
+      (SELECT concept_code_1, concept_code_2, relationship_id, vocabulary_id_2
+       FROM concept_relationship_manual crm_old
+
+       WHERE NOT exists(SELECT source_code,
+                               target_concept_code,
+                               'HCPCS',
+                               target_vocabulary_id,
+                               CASE
+                                   WHEN to_value ~* 'value' THEN 'Maps to value'
+                                   WHEN to_value ~* 'Is a' THEN 'Is a'
+                                   WHEN to_value ~* 'Subsumes' THEN 'Subsumes'
+                                   ELSE 'Maps to' END
+                        FROM dev_hcpcs.hcpcs_mapped crm_new
+                        WHERE source_code = crm_old.concept_code_1
+                          AND target_concept_code = crm_old.concept_code_2
+                          AND target_vocabulary_id = crm_old.vocabulary_id_2
+                          AND CASE
+                                  WHEN to_value ~* 'value' THEN 'Maps to value'
+                    WHEN to_value ~* 'Is a' THEN 'Is a'
+                    WHEN to_value ~* 'Subsumes' THEN 'Subsumes'
+                   ELSE 'Maps to' END = crm_old.relationship_id
+
+    )
+    AND invalid_reason IS NULL
+    )
+;
```

Comment:
Here we need to add a mapping resurrection script. Please use this example, but adopt to the _mapped table:

-- activate mapping, that became valid again
UPDATE concept_relationship_manual crm
SET invalid_reason = null,
    valid_end_date = to_date('20991231','yyyymmdd'),
    valid_start_date =current_date

--SELECT * FROM concept_relationship_manual crm --use this SELECT for QA
WHERE invalid_reason = 'D' -- activate only deprecated mappings

    AND EXISTS (SELECT 1 -- activate mapping if the same exists in the current manual file
                    FROM refresh_lookup_done rl
                    WHERE rl.icd_code = crm.concept_code_1 --the same source_code is mapped
                        AND rl.repl_by_code = crm.concept_code_2 --to the same concept_code
                        AND rl.repl_by_vocabulary = crm.vocabulary_id_2 --of the same vocabulary
                        AND rl.repl_by_relationship = crm.relationship_id --with the same relationship
        )
;